common/codex32.c

```diff
@@ -374,3 +398,48 @@ struct codex32 *codex32_decode(const tal_t *ctx,
 
 	return parts;
 }
+
+/* Returns Codex32 encoded secret of the seed provided. */
+char *codex32_secret_encode(const tal_t *ctx,
+			    const char *id,
+			    const u32 threshold,
+			    const u8 *seed,
+			    size_t seedlen) {
+
+	char *c = tal(ctx, char);
+	const struct checksum_engine *csum_engine;
+	const char *hrp = "ms";
+	assert(threshold <= 9 && threshold >= 0 &&
+		threshold != 1 && strlen(id) == 4);
+
+	/* Every codex32 has hrp `ms` and since we are generating a
+	 * secret it's share index would be `s` and threshold given by user. */
+	tal_append_fmt(&c, "%s1%d%ss", hrp, threshold, id);
+
+	uint8_t next_u5 = 0, rem = 0;
+
+        for (size_t i = 0; i < seedlen; i++) {
+            /* Each byte provides at least one u5. Push that. */
+            uint8_t u5 = (next_u5 << (5 - rem)) | seed[i] >> (3 + rem);
+
+	    tal_append_fmt(&c, "%c", bech32_charset[u5]);
+            next_u5 = seed[i] & ((1 << (3 + rem)) - 1);
+
+            /* If there were 2 or more bits from the last iteration, then
+             * this iteration will push *two* u5s. */
+            if(rem >= 2) {
+	        tal_append_fmt(&c, "%c", bech32_charset[next_u5 >> (rem - 2)]);
+                next_u5 &= (1 << (rem - 2)) - 1;
+            }
+            rem = (rem + 8) % 5;
+        }
+        if(rem > 0) {
+	    tal_append_fmt(&c, "%c", bech32_charset[next_u5 << (5 - rem)]);
+        }
+	char *codex_datastr = c+3;
+
```

Comment:
Erk, this doesn't quite work!  Because create_checksum may have to reallocate, and move the string...

We'll need to do somthing like:

char csum[csum_engine.len];
create_checksum(hrp, csum, c + 3, csum_engine);
tal_append_fmt(&c, "%.*s", (int)csum_engine.len, csum);